Adding a button to publish code as an annonymous gist

soupi opened the pull request:
Adds a 'Share' button at the top right of the screen.

Pressing on it will trigger a confirm message. after that it will:
- Create an annonymous gist via github
- Refresh the page with the `gist` query parameter updated to the newly created gist
or it will fail and `alert` the user and will print the response object in console.

A few more changes:

- Tooltips for buttons
- pointer cursor for buttons
- Don't load session if `gist` query parameter is present (does that make sense?)

[Viewable here](https://soupi.github.io/trypurescript).

paf31 commented on js/index.js:
```diff
@@ -285,16 +280,45 @@ $(function() {
       }
     });
+    var publishNewGist = function() {
+        if (!confirm('Do you really want to publish as a Gist?')) { return; }
```
Maybe add something here along the lines of "This code will be publicly visible"? And add the word "anonymous" before Gist so it's clear it's not associated to the user's account if they happen to be logged in too?

soupi replied:
sure. sounds good.

paf31 commented on js/index.js:
```diff
@@ -285,16 +280,45 @@ $(function() {
+                $.setQueryParameters($.QueryString);
+            })
+            .error( function(e) {
+                console.warn("Gist creation failed: ", e);
```
Maybe this should alert the user too (without the `e` part of course) and direct them to look at the console output for more info?

paf31 commented:
👍 Looks great, thanks!

paf31 commented:
By the way, on my screen, it wasn't obvious that share was there, until I looked for it and found it on the right. Why not keep it with the other buttons?

soupi commented:
I didn't like the way it looked much, especially since the Show JS can appear and disappear causing it to move. for the case we add more interaction with gists I thought it'd be better to group them to the right.
Also, it will probably look better after adding a few icons :)

soupi commented:
I made the following changes:

- Added an alert on failure
- Edited the text on gist creation
- Moved the share button before the compile button. does that looks better?

paf31 commented:
Looks great, thanks!
